examples/morpheusvm/tests/workload/genesis.go

```diff
@@ -72,7 +74,17 @@ func newDefaultAuthFactories() []chain.AuthFactory {
 
 func NewTestNetworkConfig(minBlockGap time.Duration) (workload.DefaultTestNetworkConfiguration, error) {
 	keys := newDefaultAuthFactories()
-	genesis := newGenesis(keys, minBlockGap)
+	testsLocalAllocations, err := tests.TestsRegistry.GenerateCustomAllocations(func() (chain.AuthFactory, error) {
+		privateKey, err := ed25519.GeneratePrivateKey()
+		if err != nil {
+			return nil, err
+		}
+		return auth.NewED25519Factory(privateKey), nil
+	})
```

Comment:
This control flow is a bit confusing. This code is used to generate the genesis, but it's not clear from here where it's going to be used even though it's coupled to the tests registry.

We are using the test registry in the integration and e2e environments where we define two entry points. This is also where we invoke `NewTestNetworkConfig`.

Could we change this function to take in a slice of genesis allocations as an argument (in addition to `minBlockGap`)?

Then we can import the test registry and use the registered custom allocations in the entry point in both `examples/morpheusvm/tests/integration/integration_test.go` and examples/morpheusvm/tests/e2e/e2e_test.go`